replica/database.hh

```diff
@@ -1424,7 +1424,7 @@ private:
     size_t max_memory_streaming_concurrent_reads() { return _dbcfg.available_memory * 0.02; }
     static constexpr size_t max_count_system_concurrent_reads{10};
     size_t max_memory_system_concurrent_reads() { return _dbcfg.available_memory * 0.02; };
-    size_t max_memory_pending_view_updates() const { return _dbcfg.available_memory * 0.1; }
+    size_t max_memory_pending_view_updates() const { return _dbcfg.available_memory * 0.04; }
```

Comment:
I'm still investigating this but the issue is that we're in fact using considerably more memory than the view backlog suggests. The baseline is that with 10% we were still getting errors with this patch and with 4% we didn't (for the included test).
For now I'm seeing that for every remote view update we have a background future, a  `abstract_view_response_handler` (relatively big compared to the view update size in the test) and also, temporarily, a second copy of the mutation as the serialized buffer that we're sending to another node, but these 3 things still don't explain most of the used up memory

Reply:
By investigating a coredump generated in a crash caused by this issue I found a more precise estimate of extra memory used for view updates, you can find some more data in  https://docs.google.com/document/d/1dKQhY7D43rNvyBlj_svTozEw_WRL05_XWoBXonv9fPI/edit?usp=sharing
With that I'm removing the change to 4% but instead I'm adding the overhead to our memory accounting. As a result, the limit should still be 10% but we should actually stop exceeding it

Reply:
@wmitros very nice work on that overhead calculation, and showing it's actually around 2200 bytes per view update, not 256 bytes. This indeed means that instead of limiting the view update backlog around 10% of memory, for small updates it could reach almost 8 times more - 80% of memory, so even without other bugs all memory could be finished even if the flow control was working as designed.

This finding should probably be backed by a new issue, and perhaps a patch to it should be done, before other things, including this PR. Your new finding is not specific to the issue fixed by this PR (which is about huge deletions causing unlimited concurrency) but is much more general, and to be honest - more important.

Reply:
@kostja did you really get upset that this PR hasn't gone in when just before the weekend @wmitros fixed the biggest problem?
I'll review this whole thing again.